wp-includes/parser/class-grammar.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+class Grammar {
+	public $rules;
+	public $rule_names;
+	public $fragment_ids;
+	public $lookahead_is_match_possible = array();
+	public $lowest_non_terminal_id;
```

Comment:
We're modeling a future WordPress public API here so let's be as restrictive about the visibility as possible, e.g. this one isn't used anywhere else so it could be private. Also, let's document these properties and look for useful names that make it all "just click" or at least have a helpful comment. A few questions I've noticed myself having after returning to this code were  "what's a fragment in fragment_id", "what's a terminal"?

Reply:
@adamziel I'm adding a TODO comment to address these later on. They seem to be all used from outside now, but maybe that's not necessary. We can look at these later more in depth.